### PR TITLE
Added redirect from generated link to change password

### PR DESCRIPTION
**Pull request by dantethedevil**

### CODE REVIEW

File: src/app/services/auth.service.ts [hunk 1]
```diff
@@ -141,4 +142,10 @@ export class AuthService {
     const decodedToken: TokenInfo = JWTDecoder(this.getToken());
     return decodedToken.jti;
   }
+  public getChangePasswordToken(): string {
```
Comment (dmitryy90) [inline src/app/services/auth.service.ts]: anotations

File: src/app/services/auth.service.ts [hunk 2]
```diff
@@ -141,4 +142,10 @@ export class AuthService {
...
+    return this.changePasswordToken;
+  }
+  public setChangePasswordToken(token: string): void {
```
Comment (dmitryy90) [inline src/app/services/auth.service.ts]: indentations

File: src/app/app.component.ts [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
 export class AppComponent implements OnInit {
-  constructor(private authService: AuthService) { }
+  constructor(private authService: AuthService, private router: Router) { }
   ngOnInit(): void {
-   this.authService.checkTokenValidity();
+    if (window.location.href.includes('resetPassword')) {
```
Comment (dmitryy90) [inline src/app/app.component.ts]: why don't you use the separate component?

File: src/app/app.component.ts [hunk 4]
```diff
@@ -1,14 +1,27 @@
...
+      const tokenIndex = window.location.href.indexOf('token');
+      const token = tokenIndex === -1 ?
+        '' :
+        window.location.href
+          .slice(tokenIndex)
+          .replace('token=', '');
```
Comment (dmitryy90) [inline src/app/app.component.ts]: too complicated way, use URLSearchParams or something similar instead

File: src/app/services/auth.service.ts (hunk 1 above)
Comment (dantethedevil) [reply src/app/services/auth.service.ts]: solved

File: src/app/services/auth.service.ts (hunk 2 above)
Comment (dantethedevil) [reply src/app/services/auth.service.ts]: solved

File: src/app/app.component.ts (hunk 4 above)
Comment (dantethedevil) [reply src/app/app.component.ts]: solved

File: src/app/app.component.ts (hunk 3 above)
Comment (dantethedevil) [reply src/app/app.component.ts]: Didn't solved, becouse we have hash and it's the only way to make redirect.

**Pull request merged**